pkg/proxy/ipvs/proxier.go

```diff
@@ -2046,6 +2046,12 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		newEndpoints.Insert(epInfo.String())
 	}
 
+	if len(newEndpoints) == 0 && onlyNodeLocalEndpoints {
+		for _, epInfo := range proxier.endpointsMap[svcPortName] {
```

Comment:
I'm wondering if we should defer this logic for when we implement `internalTrafficPolicy` https://github.com/kubernetes/kubernetes/pull/96600? This way the behavior to fall-back to cluster-wide endpoints is opt-in by user and expected

Reply:
How far ahead in time is it? This is a very small PR and may serve as a gap-filler.

Reply:
That PR is targetted for v1.21. With that PR you can set `internalTrafficPolicy: PreferLocal` which I think would achieve the same behavior as this. Is that true?

Reply:
Sounds perfect :+1: I'll apply the PT and test it when I get time, hopefully tomorrow.

Reply:
Cool. It is good to know that #96600 can solve the problem, and that PR will be in v1.21. Thanks @andrewsykim for your review and sharing, and thanks @uablrek for working with this PR.

Reply:
[Service Internal Traffic Policy is not used when externalTrafficPolicy is set to Local on a Service](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/#constraints)

Reply:
internalTrafficPolicy can't be used to solve this issue anyway. Please see the comment below.